include/zephyr/sys_clock.h

```diff
@@ -248,6 +248,24 @@ static inline bool sys_timepoint_expired(k_timepoint_t timepoint)
 	return K_TIMEOUT_EQ(sys_timepoint_timeout(timepoint), Z_TIMEOUT_NO_WAIT);
 }
 
+/**
+ * @brief Compare two timepoint values.
+ *
+ * @param a Timepoint to compare
+ * @param b Timepoint to compare against.
+ * @return true if a is is going to expire sooner than b, false otherwise.
+ */
+bool sys_timepoint_lessthan(k_timepoint_t a, k_timepoint_t b);
```

Comment:
Maybe `_before` is clearer?

```suggestion
bool sys_timepoint_before(k_timepoint_t a, k_timepoint_t b);
```